File: 프로그래머스/lv1/67256. ［카카오 인턴］ 키패드 누르기/［카카오 인턴］ 키패드 누르기.py
def solution(numbers, hand):
    answer = ''
    distance = [[1,3],[0,0],[1,0],[2,0],[0,1],[1,1],[2,1],[0,2],[1,2],[2,2],[0,3],[2,3]]
    left_location = 10
    right_location = 11

    for i in numbers:
        if i in [1,4,7]:
            answer += 'L'
            left_location = i
        elif i in [3,6,9]:
            answer += 'R'
            right_location = i
        else:
            left_dist = (abs(distance[i][0] - distance[left_location][0]) + 
                     abs(distance[i][1] - distance[left_location][1]))
            right_dist = (abs(distance[i][0] - distance[right_location][0]) + 
                     abs(distance[i][1] - distance[right_location][1]))
            if left_dist > right_dist:
                answer += 'R'
                right_location = i  
            elif left_dist < right_dist:
                answer += 'L'
                left_location = i
            else:
                if hand == 'right':
                    answer += 'R'
                    right_location = i    
                else:
                    answer += 'L'
                    left_location = i
    return answer

File: 프로그래머스/lv1/67256. ［카카오 인턴］ 키패드 누르기/［카카오 인턴］ 키패드 누르기.java
import java.util.HashMap;
import java.util.Map;

public class Solution {
    public static String solution(int[] numbers, String hand) {
        StringBuilder answer = new StringBuilder();
        int[][] distance = {{1,3},{0,0},{1,0},{2,0},{0,1},{1,1},{2,1},{0,2},{1,2},{2,2},{0,3},{2,3}};
        int leftLocation = 10;
        int rightLocation = 11;

        for (int i : numbers) {
            if (i == 1 || i == 4 || i == 7) {
                answer.append("L");
                leftLocation = i;
            } else if (i == 3 || i == 6 || i == 9) {
                answer.append("R");
                rightLocation = i;
            } else {
                int[] currentLocation = distance[i];
                int leftDistance = Math.abs(currentLocation[0] - distance[leftLocation][0])
                        + Math.abs(currentLocation[1] - distance[leftLocation][1]);
                int rightDistance = Math.abs(currentLocation[0] - distance[rightLocation][0])
                        + Math.abs(currentLocation[1] - distance[rightLocation][1]);

                if (leftDistance > rightDistance) {
                    answer.append("R");
                    rightLocation = i;
                } else if (leftDistance < rightDistance) {
                    answer.append("L");
                    leftLocation = i;
                } else {
                    if (hand.equals("right")) {
                        answer.append("R");
                        rightLocation = i;
                    } else {
                        answer.append("L");
                        leftLocation = i;
                    }
                }
            }
        }

        return answer.toString();
    }
}
